### ISSUE TITLE
Wrong formatting a line comment within type declaration in class

### ISSUE THREAD

**Issue by MichiFr** [SAP/abap-cleaner#335]
Hi!

I'm using the latested Abap-Cleaner and just let it format a global class. I noticed that the cleaner did format a line comment wrong. Instead of keeping the comment in the current line, at the beginning, the line has been moved to the end of the previous line.

I'm using the **default** profile and the code preview gives the same result, btw.

This is the code block, pls. note the line 20 ***,billing_to_partner** that is commented out.
``` 
types:
    BEGIN OF t_data_ext
            ,operation            TYPE zcl_cs_operations=>t_operation-vornr
            ,workhour             TYPE zzconfirmworkhour
            ,unit_workhour        TYPE zzconfirmworkhrunit
            ,travel               TYPE zzconfirmtravel
            ,unit_travel          TYPE zzconfirmtravelunit
            ,replace_costs        TYPE zzconfirmreplcosts
            ,unit_replace_costs   TYPE zzconfirmreplcostsunit
            ,rebate               TYPE zzconfirmrebate
            ,unit_rebate          TYPE zzconfirmrebateunit
            ,other                TYPE zzconfirmother
            ,unit_other           TYPE zzconfirmotherunit
            ,material             TYPE zzconfirmmaterial
            ,unit_material        TYPE zzconfirmaterialunit
            ,goodwill             TYPE zzconfirmgoodwill
            ,billing              TYPE zzconfirmbilling
            ,billing_to           TYPE zzbillingdecision
*,billing_to_partner TYPE ihpa-parnr
            ,warranty             TYPE zzconfirmwarranty
            ,revisited            TYPE zzconfirmrevisited
            ,plumber_informed     TYPE zzconfirmplumbinfo
            ,plumber_onsite       TYPE zzconfirmplumbonsite
            ,task_for_office      TYPE zzconfirmofficetask
            ,customer_notonsite   TYPE zzconfirmcustomernotonsite
            ,confirmation_info    TYPE zcl_cs_longtext=>tt_lines
            ,start_date           TYPE zzconfirmsdate
            ,end_date             TYPE zzconfirmedate
            ,service_render_date  TYPE zcl_cs_order_confirmation_fsm=>t_serv_rend_date
            "Weitere Meldungsdaten
            ,symptoms             TYPE tt_symptoms  "Schadensbilder
            ,parts                TYPE tt_parts     "Objektteile
            ,item_quantities      TYPE tt_item_quantities "Positionsmengen
            ,causes               TYPE tt_causes  "Ursachen
            ,spareparts           TYPE tt_spareparts "Ersatzteile
            ,in_operation_date    TYPE iflot-datab  "Inbetriebnahmedatum
            ,closing_partner      TYPE zcl_cs_order_confirmation=>t_closing_partner   "Rückmeldepartner
            ,service_order        TYPE caufvd-aufnr
            ,dialog_answers       type zcl_cs_order_confirmation_fsm=>t_dialog_answers
          ,END OF t_data_ext .
```

After running the code cleaner, starting with Ctrl+4, the result is like:
```
TYPES:
      BEGIN OF t_data_ext,
        operation           TYPE zcl_cs_operations=>t_operation-vornr,
        workhour            TYPE zzconfirmworkhour,
        unit_workhour       TYPE zzconfirmworkhrunit,
        travel              TYPE zzconfirmtravel,
        unit_travel         TYPE zzconfirmtravelunit,
        replace_costs       TYPE zzconfirmreplcosts,
        unit_replace_costs  TYPE zzconfirmreplcostsunit,
        rebate              TYPE zzconfirmrebate,
        unit_rebate         TYPE zzconfirmrebateunit,
        other               TYPE zzconfirmother,
        unit_other          TYPE zzconfirmotherunit,
        material            TYPE zzconfirmmaterial,
        unit_material       TYPE zzconfirmaterialunit,
        goodwill            TYPE zzconfirmgoodwill,
        billing             TYPE zzconfirmbilling,
        billing_to          TYPE zzbillingdecision *,billing_to_partner TYPE ihpa-parnr
              ,
        warranty            TYPE zzconfirmwarranty,
        revisited           TYPE zzconfirmrevisited,
        plumber_informed    TYPE zzconfirmplumbinfo,
        plumber_onsite      TYPE zzconfirmplumbonsite,
        task_for_office     TYPE zzconfirmofficetask,
        customer_notonsite  TYPE zzconfirmcustomernotonsite,
        confirmation_info   TYPE zcl_cs_longtext=>tt_lines,
        start_date          TYPE zzconfirmsdate,
        end_date            TYPE zzconfirmedate,
        service_render_date TYPE zcl_cs_order_confirmation_fsm=>t_serv_rend_date " Weitere Meldungsdaten
              ,
        symptoms            TYPE tt_symptoms,                                                             " Schadensbilder
        parts               TYPE tt_parts,                                                                " Objektteile
        item_quantities     TYPE tt_item_quantities,                                                      " Positionsmengen
        causes              TYPE tt_causes,                                                               " Ursachen
        spareparts          TYPE tt_spareparts,                                                           " Ersatzteile
        in_operation_date   TYPE iflot-datab,                                                             " Inbetriebnahmedatum
        closing_partner     TYPE zcl_cs_order_confirmation=>t_closing_partner,                            " Rückmeldepartner
        service_order       TYPE caufvd-aufnr,
        dialog_answers      TYPE zcl_cs_order_confirmation_fsm=>t_dialog_answers,
      END OF t_data_ext.
```

Did you spot the wrong part of the code: 
```
billing_to          TYPE zzbillingdecision *,billing_to_partner TYPE ihpa-parnr
``` 

Is this sthg. I could prevent by adjusting the profile or is it more or less a bug?

Michael

**Comment by jmgrassau** [comment]
Hi Michael, 

thanks a lot for reporting this – that's clearly a bug! (It's certainly also a very unusual way of placing commas… but that's not an excuse :-)

Kind regards,
Jörg-Michael

**Comment by MichiFr** [comment]
Yes, surely is unusual but should not confuse the clearner ;-)

Thanks for reading and probably fixing it!

**Comment by jmgrassau** [comment]
Hi Michael, 

this should be fixed with the next release: For one thing, the rule "Remove space before commas and period" will have a new option "Move comma or period across comment lines" (default on), while will also move the last comma in this example:

![image](https://github.com/SAP/abap-cleaner/assets/79639298/da633f28-1b3b-425c-bd01-0f23c1f60b43)

What can't be done, of course, is to handle the comma at the beginning of a commented-out code line, so if that's ever commented in again, some manual adjustment will be necessary (which seems to me like a fair punishment for such an awful coding style):

![image](https://github.com/SAP/abap-cleaner/assets/79639298/4958793f-c568-441d-baf2-859bbd43003f)

Secondly, even if someone deactivates the new option, the "Align declarations" rule will now keep comment lines (instead of erroneously moving them to the end of the previous line):

![image](https://github.com/SAP/abap-cleaner/assets/79639298/2d885cbd-372b-4c64-8589-beabad537454)

As you can see, the commas end up in a separate lines in this case, but at least, no syntax error is introduced; and if  "Remove space before commas and period" is active, you won't get this anyway. 

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
P.S.: So, altogether, you will get this for your example:

```ABAP
TYPES:
  BEGIN OF t_data_ext,
    operation           TYPE zcl_cs_operations=>t_operation-vornr,
    workhour            TYPE zzconfirmworkhour,
    unit_workhour       TYPE zzconfirmworkhrunit,
    travel              TYPE zzconfirmtravel,
    unit_travel         TYPE zzconfirmtravelunit,
    replace_costs       TYPE zzconfirmreplcosts,
    unit_replace_costs  TYPE zzconfirmreplcostsunit,
    rebate              TYPE zzconfirmrebate,
    unit_rebate         TYPE zzconfirmrebateunit,
    other               TYPE zzconfirmother,
    unit_other          TYPE zzconfirmotherunit,
    material            TYPE zzconfirmmaterial,
    unit_material       TYPE zzconfirmaterialunit,
    goodwill            TYPE zzconfirmgoodwill,
    billing             TYPE zzconfirmbilling,
    billing_to          TYPE zzbillingdecision,
*,billing_to_partner TYPE ihpa-parnr
    warranty            TYPE zzconfirmwarranty,
    revisited           TYPE zzconfirmrevisited,
    plumber_informed    TYPE zzconfirmplumbinfo,
    plumber_onsite      TYPE zzconfirmplumbonsite,
    task_for_office     TYPE zzconfirmofficetask,
    customer_notonsite  TYPE zzconfirmcustomernotonsite,
    confirmation_info   TYPE zcl_cs_longtext=>tt_lines,
    start_date          TYPE zzconfirmsdate,
    end_date            TYPE zzconfirmedate,
    service_render_date TYPE zcl_cs_order_confirmation_fsm=>t_serv_rend_date,
    " Weitere Meldungsdaten
    symptoms            TYPE tt_symptoms,                                     " Schadensbilder
    parts               TYPE tt_parts,                                        " Objektteile
    item_quantities     TYPE tt_item_quantities,                              " Positionsmengen
    causes              TYPE tt_causes,                                       " Ursachen
    spareparts          TYPE tt_spareparts,                                   " Ersatzteile
    in_operation_date   TYPE iflot-datab,                                     " Inbetriebnahmedatum
    closing_partner     TYPE zcl_cs_order_confirmation=>t_closing_partner,    " Rückmeldepartner
    service_order       TYPE caufvd-aufnr,
    dialog_answers      TYPE zcl_cs_order_confirmation_fsm=>t_dialog_answers,
  END OF t_data_ext.
```

**Comment by MichiFr** [comment]
**That fits perfectly well - good job!** 

Regarding the comma after the comment character, this is OK, too, for me and I guess for others, too. One could de-comment those lines, call the cleaner and comment it again afterwards. IMO, this does not make sense anyway.

**Comment by jmgrassau** [comment]
Hi Michael, 

thanks for your feedback – this should now be fixed with version 1.18.0, which was just released!

Kind regards,
Jörg-Michael

**Comment by MichiFr** [comment]
Updated -> Works! 
Perfect, thanks!
